inc/saiswitch.h

```diff
@@ -2256,6 +2297,44 @@ typedef enum _sai_switch_attr_t
      */
     SAI_SWITCH_ATTR_SUPPORTED_FAILOVER_MODE,
 
+    /**
+     * @brief Tunnel mappers for encap
+     *
+     * @type sai_object_list_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_TUNNEL_MAP
+     * @default empty
+     */
+    SAI_SWITCH_ATTR_TUNNEL_ENCAP_ECN_MAPPERS,
+
+    /**
+     * @brief Tunnel encap ECN Mode load method
+     *
+     * @type sai_switch_tunnel_encap_ecn_mode_t
+     * @flags CREATE_AND_SET
+     * @default SAI_SWITCH_TUNNEL_ENCAP_ECN_MODE_STANDARD
+     */
+    SAI_SWITCH_ATTR_TUNNEL_ENCAP_ECN_MODE,
+
+    /**
+     * @brief Tunnel mappers for decap
+     *
+     * @type sai_object_list_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_TUNNEL_MAP
+     * @default empty
+     */
+    SAI_SWITCH_ATTR_TUNNEL_DECAP_ECN_MAPPERS,
+
+    /**
+     * @brief Tunnel decap ECN Mode load method
+     *
+     * @type sai_switch_tunnel_decap_ecn_mode_t
+     * @flags CREATE_AND_SET
+     * @default SAI_SWITCH_TUNNEL_DECAP_ECN_MODE_STANDARD
+     */
+    SAI_SWITCH_ATTR_TUNNEL_DECAP_ECN_MODE,
```

Comment:
if we believe switch level configuration is needed for tunnel QoS, why do we expose it only for ECN ? and not for DSCP, TTL, etc. ?
seems incomplete model

Reply:
Current requirement is to enhance it for ECN mode only but if we all agree then yes other attributes like TTL/DSCP are good candidates as well.

Reply:
As discussed in community meeting, current requirement is specific to ECN (silicon vendor limitations). Other attributes if needed will be addressed as a cleaner encapsulate approach.